common/protob/messages-cardano.proto

```diff
@@ -58,6 +65,7 @@ message CardanoSignTx {
     repeated CardanoTxOutputType outputs = 2;   // outputs to be used in transaction
     optional uint32 transactions_count = 3;     // transactions count
     optional uint32 protocol_magic = 5;         // network's protocol magic
+    optional CardanoVersion version = 6;
```

Comment:
Here we have a problem. The data for tx signing differ markedly between Byron and Shelley. Thus it would probably be less complicated to support separate CardanoSignTxByron and CardanoSignTxShelley messages, with 4 entry points instead of 3. 

Another option is having a generic tx_data field in the protobuf msg, which would be interpreted differently in Byron and Shelley, and/or subsequent requests for additional data (as is now used for Byron via transactions_count).